e2e/test/prerequisites/E2ETestsSetup/e2eTestsSetup.ps1

```diff
@@ -0,0 +1,477 @@
+    if ($azureContext.id -ne $SubscriptionId)
+    {
+        Write-Host "`nSelecting subscription $SubscriptionId"
+        az account set --subscription $SubscriptionId
+    }
+
+    return $azureContext
+}
+
+#################################################################################################
+# Set required parameters
+#################################################################################################
+
+$Region = $Region.Replace(' ', '')
+
+$StorageAccountName = "$($ResourceGroup.ToLower())sa"
+$HubName = $ResourceGroup
+$AppRegistrationName = $ResourceGroup
+$DeviceProvisioningServiceName = $ResourceGroup
+$FarRegion = "southeastasia"
+$FarHubName = $ResourceGroup + "Far"
+$UploadCertificateName = "group1-certificate"
+
+############################################################################################################################
+# Cleanup old certs and files that can cause a conflict
+############################################################################################################################
+
+Write-Host("`nCleanup old certs and files that can cause a conflict")
+Get-ChildItem "Cert:\LocalMachine\My" | Where-Object {$_.Issuer.Contains("CN=$SubjectPrefix") } | Remove-Item
+Get-ChildItem "Cert:\LocalMachine\My" | Where-Object {$_.Issuer.Contains("CN=xdevice1") } | Remove-Item
+Get-ChildItem "Cert:\LocalMachine\My" | Where-Object {$_.Issuer.Contains("CN=iothubx509device1") } | Remove-Item
+Get-ChildItem .\ | Where-Object  {$_.Name.EndsWith(".pfx") } | Remove-Item
+Get-ChildItem .\ | Where-Object  {$_.Name.EndsWith(".cer") } | Remove-Item
+Get-ChildItem .\ | Where-Object  {$_.Name.EndsWith(".p7b") } | Remove-Item
+
+########################################################################################################
+# Generate self signed certs and to use in DPS and IOT Hub
+# New certs will be generated each time you run the script as the script cleans up in the end
+########################################################################################################
+
+Write-Host "`nGenerating self singed certs"
+
+$SubjectPrefix = "Azure IoT Test"
+$RootCommonName = "$SubjectPrefix Test Root CA"
+$IntermediateCert1CommonName = "$SubjectPrefix Intermediate 1 CA"
+$IntermediateCert2CommonName = "$SubjectPrefix Intermediate 2 CA"
+$GroupCertCommonName = "xdevice1"
+$DeviceCertCommonName = "iothubx509device1"
+$IotHubCertCommonName = "iothubx509device1"
+
+$RootCertPath = "./Root.cer"
+$IndividualDeviceCertPath = "./Device.cer"
+$verificationCertPath = "./verification.cer"
+
+$GroupPfxPath = "./Group.pfx"
+$IndividualDevicePfxPath = "./Device.pfx"
+$IotHubPfxPath = "./IotHub.pfx"
+
+$GroupCertChainPath = "./GroupCertChain.p7b"
+
+# Generate self signed Root and Intermediate CA cert, expiring in 2 years
+# These certs are used for signing so ensure to have the correct KeyUsage - CertSign and TestExtension - ca=TRUE&pathlength=12
+
+$rootCACert = New-SelfSignedCertificate `
```

Comment:
If I run this script once for my own e2e env, and then run it again for our pipeline, the second run will overwrite my certs, right? Do I need to keep these certs long term, or are they already stored in key vault?

If stored in keyvault, perhaps we delete the cert files after.

Reply:
We cleanup all certs in the end as we don't need it on our machine so it should be ok. And the E2E tests resources will be created in a different RG than your own so it won't overwrite your KV values. Also we should either run as the SP that runs the tests to create resources for our pipeline or give the permissions to it after.

Reply:
There is still some improvements we can make. Currently I'm just storing the value of the pfx. We could however store them as certs but it is some extra work to get the byte64 and put it in the evn variable.